File: server/src/main/java/org/hkurh/doky/security/DokyUserDetails.java
package org.hkurh.doky.security;

import org.hkurh.doky.entities.UserEntity;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Collections;

public class DokyUserDetails implements UserDetails {
    private String username;
    private String password;
    private Collection<? extends GrantedAuthority> grantedAuthorities;

    public static DokyUserDetails createUserDetails(final UserEntity userEntity) {
        var dokyUserDetails = new DokyUserDetails();
        dokyUserDetails.username = userEntity.getUid();
        dokyUserDetails.password = userEntity.getPassword();
        dokyUserDetails.grantedAuthorities = Collections.singletonList(DokyAuthority.ROLE_USER);

        return dokyUserDetails;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return grantedAuthorities;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: server/src/main/java/org/hkurh/doky/security/DokyUserDetails.kt
package org.hkurh.doky.security

import org.hkurh.doky.users.UserEntity
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class DokyUserDetails : UserDetails {
    private var userEntity: UserEntity? = null
    private var username: String = ""
    private var password: String = ""
    private var grantedAuthorities: Collection<GrantedAuthority?>? = null

    override fun getAuthorities(): Collection<GrantedAuthority?> = grantedAuthorities!!
    override fun getPassword(): String = password
    override fun getUsername(): String = username
    override fun isAccountNonExpired(): Boolean = true
    override fun isAccountNonLocked(): Boolean = true
    override fun isCredentialsNonExpired(): Boolean = true
    override fun isEnabled(): Boolean = true
    fun getUserEntity(): UserEntity? = userEntity

    companion object {
        fun createUserDetails(userEntity: UserEntity): DokyUserDetails {
            val dokyUserDetails = DokyUserDetails()
            dokyUserDetails.userEntity = userEntity
            dokyUserDetails.username = userEntity.uid
            dokyUserDetails.password = userEntity.password
            dokyUserDetails.grantedAuthorities = listOf(DokyAuthority.ROLE_USER)
            return dokyUserDetails
        }
    }
}
